src/strategies.rs

```diff
@@ -38,12 +104,86 @@ pub trait BuildStrategy {
     fn login(&self) -> Result<()>;
 }
 
-pub fn determine_build_strategy(
-    uuid: Uuid,
-    creds: Option<Credentials>,
-    oci_required: bool,
-) -> Result<Rc<dyn BuildStrategy>> {
-    trace!("BuildStrategy::determine_strategy({uuid})");
+/// Allows agnostic inspection of images.
+pub trait InspectStrategy: Sync + Send {
+    fn get_labels(&self, image_name: &str, tag: &str) -> Result<ImageInspection>;
+}
+
+/// Gets the current build's UUID
+pub fn get_build_id() -> Uuid {
+    *BUILD_ID
+}
+
+/// Gets the current run's build strategy
+pub fn get_build_strategy() -> Arc<dyn BuildStrategy> {
+    BUILD_STRATEGY.clone()
+}
+
+/// Gets the current run's inspectioin strategy
+pub fn get_inspection_strategy() -> Arc<dyn InspectStrategy> {
+    INSPECT_STRATEGY.clone()
+}
+
+/// Retrieve the `os_version` for an image.
+///
+/// This gets cached for faster resolution if it's required
+/// in another part of the program.
+pub fn get_os_version(recipe: &Recipe) -> Result<String> {
+    trace!("get_os_version({recipe:#?})");
+    let image = format!("{}:{}", &recipe.base_image, &recipe.image_version);
+
+    let mut os_version_lock = OS_VERSION
+        .lock()
+        .map_err(|e| anyhow!("Unable set OS_VERSION {e}"))?;
+
+    let entry = os_version_lock.get(&image);
+
+    let os_version = match entry {
+        None => {
+            let inspection =
+                INSPECT_STRATEGY.get_labels(&recipe.base_image, &recipe.image_version)?;
+
+            let os_version = inspection.get_version().unwrap_or_else(|| {
+                warn!("Version label does not exist on image, using version in recipe");
+                recipe.image_version.to_string()
+            });
```

Comment:
Now that we have a way to dynamically choose an inspection strategy that supports not having skopeo, I'm thinking we should just error if the label for the os version doesn't exist on the image.